test/unit/driver-specs.js

```diff
@@ -0,0 +1,95 @@
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import 'mochawait';
+//import ADB from 'appium-adb';
+//import { withMocks } from 'appium-test-support';
+import { SelendroidDriver } from '../../lib/driver.js';
+import sinon from 'sinon';
+import path from 'path';
+//import { BaseDriver } from 'appium-base-driver';
+
+chai.should();
+chai.use(chaiAsPromised);
+
+describe('driver.js', () => {
+//  let adb = new ADB();
+
+
+  describe('constructor', () => {
+    it('calls BaseDriver constructor with opts', () => {
+      let driver = new SelendroidDriver({foo: 'bar'});
+      driver.should.exist;
+      driver.opts.foo.should.equal('bar');
+    });
+  });
+
+  describe('createSession', () => {
+    it('should throw an error if app can not be found', async () => {
+      let driver = new SelendroidDriver({}, false);
+      await driver.createSession({app: 'foo.apk'}).should.be.rejectedWith('app apk');
+    });
+
+    it('should set sessionId', async () => {
+      let driver = new SelendroidDriver({}, false);
+      sinon.mock(driver).expects('checkAppPresent')
+                        .once()
+                        .returns(Promise.resolve());
+      sinon.mock(driver).expects('startSelendroidSession')
+                        .once()
+                        .returns(Promise.resolve());
+      await driver.createSession({cap: 'foo'});
+
+      driver.sessionId.should.exist;
+      driver.caps.cap.should.equal('foo');
+    });
+  });
+
+  describe('checkAppPresent', async () => {
+    it('should resolve if app present', async () => {
+      let driver = new SelendroidDriver({}, false);
+      sinon.mock(driver).expects('checkAppPresent')
+                        .returns(Promise.resolve());
+      sinon.mock(driver).expects('startSelendroidSession')
+                        .returns(Promise.resolve());
+
+      await driver.createSession({app: path.resolve('.')});
+
+      driver.checkAppPresent.restore();
```

Comment:
I had the restore, because `checkAppPresent` is what we're testing in this test, but Im mocked it to create the session. I guess both the mock and the restore aren't necessary because it should find it :) removing those.

In the next test, `checkAppPresent` has to be mocked to return `true` in order to create the session, but we have to restore it before actually testing it :)

I'm not sure about the verify() calls.. these tests fail properly without calling `verify` and we don't need to restore because I'm not using `withMocks`